scripts/modelscope-text2vid.py

```diff
@@ -63,13 +71,76 @@ def process(skip_video_creation, ffmpeg_location, ffmpeg_crf, ffmpeg_preset, fps
         print('Starting text2video')
         print('Pipeline setup')
 
-        if pipe is None:
-            pipe = setup_pipeline()
+        # optionally store pipe in global between runs
+        if keep_pipe:
+            if savedPipe is None:
+                savedPipe = setup_pipeline()
+            pipe = savedPipe
+        else:
+            if savedPipe is None:
+                pipe = setup_pipeline()
+            else:
+                pipe = savedPipe
+                savedPipe = None
+
+        device=devices.get_optimal_device()
+        print('device',device)
+
+
+        if do_img2img and img2img_frames:
+            print("loading frames")
+            pattern = os.path.join(img2img_frames, '[0-9][0-9][0-9][0-9][0-9].png')
+            matching_files = glob.glob(pattern)[:frames]
+            images=[]
+            for file in matching_files:
+                image=Image.open(file)
+                image=image.resize((height,width), Image.ANTIALIAS)
+                array = np.array(image)
+                images+=[array]
+
+            #print(images)
+
+            images=np.stack(images)# f h w c
+            batches=1
+            n_images=np.tile(images[np.newaxis, ...], (batches, 1, 1, 1, 1)) # n f h w c
+            bcfhw=n_images.transpose(0,4,1,2,3)
+            #convert to 0-1 float
+            bcfhw=bcfhw.astype(np.float32)/255
+            bfchw=bcfhw.transpose(0,2,1,3,4)#b c f h w
+
+            print("got here!",bfchw.shape)
+
+            vd_out=torch.from_numpy(bcfhw).to("cuda")
+
+            #should be -1,1, not 0,1
+            vd_out=2*vd_out-1
+
+            #images should have shape # ncfhw (batches, channels [3], frames, height, width)
+            #and might have to be autoencoded in batches
+
+            #latent_h, latent_w = height // 8, width // 8
+            #latents should have shape num_sample, 4, max_frames, latent_h,latent_w
+            print("computing latents")
+            latents = pipe.compute_latents(vd_out).to(device)
+
+            #noise=torch.rand_like(latents)
+            latent_h, latent_w = height // 8, width // 8
+            noise=torch.randn(1, 4, frames, latent_h,
+                                          latent_w).to(device)
+
+
+            latents=latents*(1-img2img_noise)+noise*img2img_noise
```

Comment:
I suspect this line is wrong, but the commented out line below it produces even worse results